### PR TITLE
chore(web): step and variant actions in the production environment

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce?

In the production environment:

1. Step and Variant node actions: 
  - edit
  - view existing conditions - if conditions exist

2. Step and Variant Editor header actions:
  - view existing conditions - if conditions exist
  - close

### Other information (Screenshots)

https://github.com/novuhq/novu/assets/2607232/e2e258b4-70cc-4f44-bf29-3f8f1859ac52

### CODE REVIEW

File: apps/web/src/pages/templates/components/EditorSidebarHeaderActions.tsx [hunk 1]
```diff
@@ -114,40 +114,44 @@ export const EditorSidebarHeaderActions = () => {
     setIsDeleteModalOpened(false);
   };
 
+  const conditionAction = isReadonly ? 'View' : hasNoFilters ? 'Add' : 'Edit';
+
   return (
     <>
       <Group noWrap spacing={12} ml={'auto'} sx={{ alignItems: 'flex-start' }}>
-        <When truthy={isAddVariantActionAvailable}>
+        <When truthy={isAddVariantActionAvailable && !isReadonly}>
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/EditorSidebarHeaderActions.tsx:122]: add variant not available in production env

File: apps/web/src/pages/templates/components/EditorSidebarHeaderActions.tsx [hunk 2]
```diff
@@ -114,40 +114,44 @@ export const EditorSidebarHeaderActions = () => {
...
           <ActionButton
             tooltip="Add variant"
             onClick={onAddVariant}
             Icon={VariantPlus}
             data-test-id="editor-sidebar-add-variant"
           />
         </When>
-        <When truthy={hasNoFilters}>
+        <When truthy={hasNoFilters && !isReadonly}>
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/EditorSidebarHeaderActions.tsx:130]: add conditions action is not available in production env

File: apps/web/src/pages/templates/components/EditorSidebarHeaderActions.tsx [hunk 3]
```diff
@@ -114,40 +114,44 @@ export const EditorSidebarHeaderActions = () => {
...
           <ActionButton
-            tooltip={`Add ${isUnderVariantsListPath ? 'group' : ''} conditions`}
+            tooltip={`${conditionAction} ${isUnderVariantsListPath ? 'group' : ''} conditions`}
             onClick={() => setConditionsOpened(true)}
             Icon={PlusIcon}
             data-test-id="editor-sidebar-add-conditions"
           />
         </When>
         <When truthy={!hasNoFilters}>
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/EditorSidebarHeaderActions.tsx:138]: when there are conditions we do allow to view them in the production env

File: apps/web/src/pages/templates/components/EditorSidebarHeaderActions.tsx [hunk 4]
```diff
@@ -114,40 +114,44 @@ export const EditorSidebarHeaderActions = () => {
...
             data-test-id="editor-sidebar-edit-conditions"
           />
         </When>
-        <ActionButton
-          tooltip={`Delete ${isUnderVariantPath ? 'variant' : 'step'}`}
-          onClick={openDeleteModal}
-          Icon={Trash}
-          data-test-id="editor-sidebar-delete"
-        />
+        <When truthy={!isReadonly}>
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/EditorSidebarHeaderActions.tsx:147]: delete action is not available in the production env

File: apps/web/src/pages/templates/editor/TemplateEditorPage.tsx [hunk 5]
```diff
@@ -81,7 +87,7 @@ function BaseTemplateEditorPage() {
           </ReactFlowProvider>
         </form>
       </PageContainer>
-      <BlueprintModal />
+      {shouldRenderBlueprintModal && <BlueprintModal />}
```
Comment (LetItRock) [inline apps/web/src/pages/templates/editor/TemplateEditorPage.tsx:90]: Do not render the blueprint modal and don't call query hooks when there is no `blueprintId` set.

I've noticed this issue in the tests, like when env is switched we do call refetch all queries, and blueprint queries do fail which results in that the page is navigated from the editor to the workflows, causing tests to fail.

File: apps/web/src/pages/templates/workflow/workflow/node-types/WorkflowNodeActions.tsx [hunk 6]
```diff
@@ -134,58 +148,56 @@ export const WorkflowNodeActions = ({
...
-                  onDelete();
-                }}
-                icon={<Trash width="16px" height="16px" />}
-                data-test-id="delete-step-action"
-              >
-                {VARIANT_TYPE_TO_DELETE[nodeType]}
-              </Dropdown.Item>
-            )}
-          </Dropdown>
+          <When truthy={!isReadOnly}>
```
Comment (LetItRock) [inline apps/web/src/pages/templates/workflow/workflow/node-types/WorkflowNodeActions.tsx:160]: don't show dropdown actions in the production env

File: apps/web/src/pages/templates/workflow/workflow/node-types/WorkflowNodeActions.tsx [hunk 7]
```diff
@@ -110,17 +119,22 @@ export const WorkflowNodeActions = ({
   }
 
   const isDelayedStep = DELAYED_STEPS.includes(channelType);
+  const conditionsAction = isReadOnly
+    ? VARIANT_TYPE_TO_VIEW_CONDITIONS[nodeType]
+    : conditionsCount > 0
+    ? VARIANT_TYPE_TO_EDIT_CONDITIONS[nodeType]
+    : VARIANT_TYPE_TO_ADD_CONDITIONS[nodeType];
```
Comment (LetItRock) [inline apps/web/src/pages/templates/workflow/workflow/node-types/WorkflowNodeActions.tsx:126]: in the production env show the tooltip text for the conditions like: `View conditions`

**Review by rifont — APPROVED** [review]
Would love for us to chat a bit more about using accessible test queries, like `cy.getByRole`. They enforce accessibility out of the box, and make tests stronger by depending on features rather than an implementation detail.

In the meantime, looks great!

**Comment by LetItRock** [comment]
> Would love for us to chat a bit more about adopting accessible test queries, like `cy.getByRole` (e.g., see [React Testing Library](https://testing-library.com/docs/react-testing-library/cheatsheet/#queries)). They enforce accessibility out of the box, and make tests stronger by depending on features rather than an implementation detail like `data-test-id`.
> 
> In the meantime, looks great!

great! I am glad that you pointed it out, totally agree with you 🙌 I always wanted to start doing that, but it requires the implementation to be semantically and i11y correct from the beginning.

**Pull request merged**